metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -631,6 +647,42 @@ object MetalsEnrichments
 
     def lineAtIndex(index: Int): Int =
       indexToLspPosition(index).getLine
+//
+//    /**
+//     * @param pos the position at which non-blank text starts
+//     * @return the length of indentation at the beginning of this string
+//     */
+//    def indentationLength(pos: Position): Int = {
```

Comment:
Why won't we use this one instead of the Scanner actually? It's similar to what I suggested beneath.

Reply:
That code is buggy! Kamil made a bug free version! But it is also much longer and more difficult to maintain!